pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+# coding=utf-8
+"""Test the API endpoints `OSTree`_ `repositories`_.
+
+This module assumes that the tests in
+:mod:`pulp_smash.tests.platform.api_v2.test_repository` hold true. The
+following trees of assumptions are explored in this module::
+
+    It is possible to create an OSTree repo with feed (CreateTestCase).
+    ├── When valid feed and branch are given repo synces successfully without
+    │   reported errors (CreateValidFeedTestCase).
+    ├── When Invalid feed or branch is given repo sync fails with errors
+    │   reported (SyncInvalidFeedTestCase).
+    └── It is possible to sync repository, copy its content to a second
+        repository, update repository metadata, sync again, add distributor
+        to the first repository and publish it and access published content
+        (PublishTestCase).
+
+    It is possible to create repository without feed (CreateTestCase).
+    └── Running sync on this repository fails with errors reported
+        (SyncWithoutFeedTestCase).
+
+.. _OSTree:
+    http://pulp-ostree.readthedocs.org/en/latest/
+.. _repositories:
+   http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/cud.html
+"""
+
+from __future__ import unicode_literals
+
+from itertools import chain
+from packaging.version import Version
+
+from pulp_smash import api, utils, selectors
+from pulp_smash.config import get_config
+from pulp_smash.constants import REPOSITORY_PATH
+from unittest2 import TestCase
+from urlparse import urljoin
+
+_VALID_FEED = 'http://dl.fedoraproject.org/pub/fedora/linux/atomic/21/'
+_VALID_BRANCHES = tuple((
+    'fedora-atomic/f21/x86_64/updates/docker-host',
+    'fedora-atomic/f21/x86_64/updates-testing/docker-host',
+))
+
+
+class _BaseTestCase(TestCase):
+    """Provide a server config, and tear down created resources."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Provide a server config and an iterable of resources to delete."""
+        cls.cfg = get_config()
+        cls.resources = set()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete created resources."""
+        client = api.Client(cls.cfg)
+        for resource in cls.resources:
+            client.delete(resource)
+
+
+def _gen_ostree_repo_body():
+    """Return OSTree repo body."""
+    return {
+        'id': utils.uuid4(),
+        'importer_type_id': 'ostree_web_importer',
+        'importer_config': {},
+        'distributors': [],
+        'notes': {'_repo-type': 'OSTREE'},
+    }
+
+
+class CreateTestCase(_BaseTestCase):
+    """Create two OSTree repositories, with and without feed."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two repositories."""
+        super(CreateTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        cls.bodies = tuple((_gen_ostree_repo_body() for _ in range(2)))
+        cls.bodies[1]['importer_config'] = {'feed': utils.uuid4()}
+        # RPM plugin doesn't validate url, puppet plugin does
```

Comment:
It's more note than a comment, to point out that behavior here is not unified. But maybe it would be more appropriate to create an issue.